R/get-path-from-object.R

```diff
@@ -158,32 +158,188 @@ get_model_id.bbi_model <- function(.mod) {
+#'
+#' @param .bbi_object The object to query. Could be
+#' a `bbi_{.model_type}_model` object,
+#' `bbi_{.model_type}_summary` object,
+#' or a tibble of class `bbi_log_df`.
+#' @param .check_exists If `TRUE`, the default, will throw an error if the file does not exist
+#' @param pull_from_config If `TRUE`, pull the data path from `bbi.json` created
+#'  during model submission, and compare to the path defined in the control stream.
 #' @param ... Arguments passed through to methods. (Currently none, but may have
 #'   some modifier arguments other model types in the future.)
-#' @importFrom fs path_rel
+#'
+#' @details
+#'
+#' ### Notes on model extensions:
+#' `bbi` has different handling for `.mod` vs `.ctl` extensions for `NONMEM`
+#' models. When a model such as `001.ctl` is targeted with `bbi`, `bbi` will
+#' create a new directory called `001`, copy the model into it (with a `.mod`
+#' extension), and do the execution directory. For more details, see `bbi`'s
+#' [documentation](https://github.com/metrumresearchgroup/bbi/blob/main/integration/markdown/expectations.md?plain=1#L19:L21)
+#'
+#'
 #' @return Absolute path to input data file
 #' @export
-get_data_path <- function(.mod, ...) {
+get_data_path <- function(
+    .bbi_object,
+    .check_exists = TRUE,
+    pull_from_config = FALSE,
+    ...
+){
   UseMethod("get_data_path")
 }
 
-#' @describeIn get_data_path Takes `bbi_{.model_type}_model` or `bbi_{.model_type}_summary` object
+#' @rdname get_data_path
 #' @export
-get_data_path.bbi_model <- function(.mod, ...) {
+get_data_path.bbi_nonmem_model <- function(
+    .bbi_object,
+    .check_exists = TRUE,
+    pull_from_config = FALSE,
+    ...
+){
+  get_data_path_nonmem(.bbi_object, .check_exists, pull_from_config, ...)
+}
+
+#' @rdname get_data_path
+#' @export
+get_data_path.bbi_nonmem_summary <- function(
+    .bbi_object,
+    .check_exists = TRUE,
+    pull_from_config = FALSE,
+    ...
+){
+  get_data_path_nonmem(.bbi_object, .check_exists, pull_from_config, ...)
+}
+
+#' @rdname get_data_path
+#' @export
+get_data_path.bbi_log_df <- function(
+    .bbi_object,
+    .check_exists = TRUE,
+    pull_from_config = FALSE,
+    ...
+){
+  data_paths <- map_chr(.bbi_object[[ABS_MOD_PATH]], function(.path) {
+    .mod <- read_model(.path)
+    get_data_path_nonmem(.mod, .check_exists, pull_from_config, ...)
+  })
+
+  return(data_paths)
+}
+
+
+#' @keywords internal
+get_data_path_nonmem <- function(
+    .bbi_object,
+    .check_exists = TRUE,
+    pull_from_config = FALSE,
+    ...
+){
+
+  if(inherits(.bbi_object, NM_SUM_CLASS)){
+    .bbi_object <- read_model(.bbi_object$absolute_model_path)
+  }
+  check_model_object(.bbi_object, "bbi_nonmem_model")
+  mod_path <- get_model_path(.bbi_object)
+
+  # Get relative data path
+  data_path_rel <- get_data_path_from_ctl(.bbi_object)
+
+  # Handling for `.mod` extensions
+  if(grepl("(?i)ctl", fs::path_ext(mod_path))){
```

Comment:
`get_model_path()` uses `find_nonmem_model_file_path()` underneath, and that's case sensitive, so there's no reason to be case insensitive here and this condition could just be

```
identical(fs::path_ext("mod_path"), "ctl")
```

(Same applies to line 269 below.)